### DISCUSSION TITLE
ReplicaSet - emulate replication lag

### DISCUSSION THREAD

**Discussion by konstantinzelinsky** [typegoose/mongodb-memory-server#864] [Q&A]
## Versions

- NodeJS: 18.19.1
- mongodb-memory-server-*: 9.1.1
- mongodb(the binary version): 5.0.3
- mongodb(the js package): -
- mongoose: 6.12.7
- system: MacOS 14.3.1

package: mongo-memory-server 

## What is your question?

We are using this library for the e2e-testing on our project and want to detect potential problems with replication lag. Is there any way we can emulate a large replication delay? Have already tried to use [secondaryDelaySecs](https://www.mongodb.com/docs/manual/reference/replica-configuration/#mongodb-rsconf-rsconf.members-n-.secondaryDelaySecs) as a configuration option for the replica instance. But it's supported and works only with [hidden instances](https://www.mongodb.com/docs/manual/core/replica-set-delayed-member/#considerations).
Maybe there is some other solution to use this library and configure replication lag?

**Comment by hasezoey** [comment]
i am personally not aware of any other way than those `secondaryDelaySecs` option

**Comment by konstantinzelinsky** [comment]
We found solution with using [fsyncLock](https://www.mongodb.com/docs/manual/reference/method/db.fsyncLock/) functionality.
We are locking all replicas instances during test, and this emulates replication lag for us
